filebeat/input/filestream/input_test.go

```diff
@@ -0,0 +1,332 @@
+	if err != nil {
+		t.Fatalf("cannot create temporary folder for test file: %+v", err)
+	}
+	defer os.RemoveAll(tmpDir)
+
+	testlogPath := filepath.Join(tmpDir, "test.log")
+	err = ioutil.WriteFile(testlogPath, []byte("first log line\n"), 0644)
+	if err != nil {
+		t.Fatalf("cannot write log lines to test file: %+v", err)
+	}
+
+	testStore := openTestStatestore(t, tmpDir)
+	inp := getInputFromConfig(
+		t,
+		testStore,
+		map[string]interface{}{
+			"paths":                             []string{testlogPath},
+			"prospector.scanner.check_interval": "24h",
+			"close.reader.on_eof":               "true",
+		},
+	)
+
+	ctx, cancelInput := context.WithCancel(context.Background())
+	inputCtx := input.Context{Logger: logp.L(), Cancelation: ctx}
+	pipeline := &mockPipelineConnector{}
+
+	var wg sync.WaitGroup
+	go func() {
+		wg.Add(1)
+		defer wg.Done()
+
+		inp.Run(inputCtx, pipeline)
+	}()
+
+	// first event has made it successfully
+	waitUntilEventCount(1, pipeline)
+	// check registry
+	checkOffsetInRegistry(t, testStore, testlogPath, len([]byte("first log line\n")))
+
+	// the second log line will not be picked up as scan_interval is set to one day.
+	err = ioutil.WriteFile(testlogPath, []byte("first log line\nsecond log line\n"), 0644)
+	if err != nil {
+		t.Fatalf("cannot write log lines to test file: %+v", err)
+	}
+	time.Sleep(1 * time.Second)
+
+	// only one event is read
+	waitUntilEventCount(1, pipeline)
+
+	cancelInput()
+	wg.Wait()
+
+	checkOffsetInRegistry(t, testStore, testlogPath, len([]byte("first log line\n")))
+}
+
+func checkOffsetInRegistry(t *testing.T, store loginp.StateStore, filepath string, expectedOffset int) {
+	fi, err := os.Stat(filepath)
+	if err != nil {
+		t.Fatalf("cannot stat file when cheking for offset: %+v", err)
+	}
+
+	identifier, _ := newINodeDeviceIdentifier(nil)
+	src := identifier.GetSource(loginp.FSEvent{Info: fi, Op: loginp.OpCreate, NewPath: filepath})
+	entry := getRegistryState(t, store, src.Name())
+
+	require.Equal(t, expectedOffset, entry.Cursor.Offset)
+}
+
+func getInputFromConfig(t *testing.T, s loginp.StateStore, c map[string]interface{}) input.Input {
+	manager := &loginp.InputManager{
+		Logger:     logp.L(),
+		StateStore: s,
+		Type:       pluginName,
+		Configure:  configure,
+	}
+
+	inp, err := manager.Create(common.MustNewConfigFrom(c))
+	if err != nil {
+		t.Fatalf("cannot create filestream input: %+v", err)
+	}
+	return inp
+}
+
+type registryEntry struct {
+	Cursor struct {
+		Offset int `json:"offset"`
+	} `json:"cursor"`
+}
+
+func getRegistryState(t *testing.T, s loginp.StateStore, key string) registryEntry {
+	inputStore, _ := s.Access()
+
+	var e registryEntry
+	err := inputStore.Get(key, &e)
+	if err != nil {
+		t.Fatalf("error when getting expected key '%s' from store: %+v", key, err)
+	}
+
+	return e
+}
+
+func waitUntilEventCount(count int, pipeline *mockPipelineConnector) {
+	for {
+		for _, c := range pipeline.clients {
+			if len(c.GetEvents()) == count {
```

Comment:
Can we somehow pass theclient directly?

The function waits until there is at least one client that has published `count` events. Given the function named I would assumed that the total number of events is summed up like:

```
sum := 0
for _, c := range pipeline.clients {
  sum += len(c.GetEvents())
}
if sum == count {
  return...
}
```

The original python tests can not distinguish by Client and compute the total number of events. The wait function(s) should behave more similar to the python tests in order to reduce the chance of errors when translating the tests.